PowerPointLabs/PowerPointLabs/PasteLab/PasteToFillSlide.cs

```diff
@@ -15,23 +19,35 @@ public static void Execute(PowerPointSlide slide, ShapeRange pastingShapes, floa
                 return;
             }
 
-            Shape shapeToFillSlide = pastingShapes[1];
+            Shape pastingShape = pastingShapes[1];
             if (pastingShapes.Count > 1)
             {
-                shapeToFillSlide = pastingShapes.Group();
+                pastingShape = pastingShapes.Group();
             }
-            shapeToFillSlide.LockAspectRatio = Microsoft.Office.Core.MsoTriState.msoTrue;
 
-            PPShape ppShapeToFillSlide = new PPShape(shapeToFillSlide);
+            // Temporary house the latest clipboard shapes
+            ShapeRange origClipboardShapes = ClipboardUtil.PasteShapesFromClipboard(slide);
```

Comment:
Is this part restoring the clipboard? Maybe you can use ClipboardUtil.RestoreClipboardAfterAction?

Reply:
Nope it is not restoring the clipboard, hence RestoreClipboardAfterAction would not work here